spec/models/manageiq/providers/amazon/aws_stubs.rb

```diff
@@ -3,22 +3,51 @@ def scaling_factor
     @data_scaling || try(:data_scaling) || 1
   end
 
-  def test_counts
+  def disconnect_inv_factor
+    # The entities like VM are disconnected instead of deleted, for comparing, we set how many was disconnected and
+    # we add them to the total count
+    @disconnect_inv_count || 0
+  end
+
+  def test_counts(scaling = nil)
+    scaling ||= scaling_factor
     {
-      :load_balancer_count                             => scaling_factor * 20,
-      :instance_vpc_count                              => scaling_factor * 20,
-      :instance_ec2_count                              => scaling_factor * 20,
-      :load_balancer_instances_count                   => scaling_factor * 10,
-      :vpc_count                                       => scaling_factor * 20,
-      :subnet_count                                    => scaling_factor * 20,
-      :network_port_count                              => scaling_factor * 20,
-      :floating_ip_count                               => scaling_factor * 20,
-      :security_group_count                            => scaling_factor * 20,
-      :inbound_firewall_rule_per_security_group_count  => scaling_factor * 5,
-      :outbound_firewall_rule_per_security_group_count => scaling_factor * 5,
+      :load_balancer_count                             => scaling * 20,
+      :availability_zone_count                         => scaling * 5,
+      :instance_vpc_count                              => scaling * 20,
+      :instance_ec2_count                              => scaling * 20,
+      :image_count                                     => scaling * 20,
+      :key_pair_count                                  => scaling * 20,
+      :stack_count                                     => scaling * 20,
+      :stack_resource_count                            => scaling * 20,
+      :stack_parameter_count                           => scaling * 20,
+      :stack_output_count                              => scaling * 20,
+      :load_balancer_instances_count                   => scaling * 10,
+      :vpc_count                                       => scaling * 20,
+      :subnet_count                                    => scaling * 20,
+      :network_port_count                              => scaling * 20,
+      :floating_ip_count                               => scaling * 20,
+      :security_group_count                            => scaling * 20,
+      :inbound_firewall_rule_per_security_group_count  => scaling * 5,
+      :outbound_firewall_rule_per_security_group_count => scaling * 5,
     }
   end
 
+  def assert_do_not_delete
```

Comment:
hm, it's really used only for the stubbed specs. Can we do that later? I have more refactorings, that could use extracting of asserts for stubbs